pydoctor/templatewriter/summary.py

```diff
@@ -26,15 +26,31 @@ def moduleSummary(module: model.Module, page_url: str) -> Tag:
         r(class_='private')
     if not isinstance(module, model.Package):
         return r
-    contents = [m for m in module.contents.values()
-                if isinstance(m, model.Module) and m.isVisible]
+    contents = list(module.submodules())
     if not contents:
         return r
-    ul = tags.ul()
+
     def fullName(obj: model.Documentable) -> str:
         return obj.fullName()
-    for m in sorted(contents, key=fullName):
-        ul(moduleSummary(m, page_url))
+
+    ul = tags.ul()
+
+    if len(contents) > 50 and not any(any(s.submodules()) for s in contents):
+        # If there are more than 50 modules and no submodule has
+        # further submodules we use a more compact presentation.
+        li = tags.li(class_='compact-modules')
+        for m in sorted(contents, key=fullName):
+            span = tags.span()
+            span(tags.code(epydoc2stan.taglink(m, m.url, label=m.name)))
+            span(', ')
+            if m.isPrivate:
+                span(class_='private')
```

Comment:
I updated the PR to add the `private` class for private modules, which is necessary because otherwise they're always shown.

Reply:
Thanks, I did not see that bug. It looks like the whole span will be hidden though. Where only one tag link and the commas should be private I think.

Reply:
Yes it it's intentional that the whole span is hidden because otherwise there would be ,,,,,, left over.

Reply:
Sorry it’s my fault I though is was one span for the whole thing.